### DISCUSSION TITLE
Operating system

### DISCUSSION THREAD

**Discussion by eorojas** [frankcohen/ReflectionsOS#10] [Ideas]
My understanding of Arduino is that it is a nice prototype environment but not a real OS suitable for a product.
What is your thinking?

I know that FreeRTOS is available for the ESP32.  

I don't have a lot of bandwidth to help, but if you decide to go that way at some point, let me know.  I did a FreeRTOS project a couple of years ago.  It is fairly straightforward.

**Comment by frankcohen** [comment]
Thanks, I'm glad for your post, glad for you to participate. Arduino's not an OS, it is a firmware environment for building C/C++ and Python applications, and Arduino takes care of sharing libraries for hardware and software related to the boards. For this project the board is an ESP32-S3-Mini and the language is C/C+. Since I'm planning to build multimedia entertainment experiences, I want to try for a cooperative multitasking operating system at first to make sure the bits arrive at the video display on time. However, that shouldn't stop you from experimenting with FreeRtos. It looks great and I'd be glad for you to experiment with it on Reflections. Frank

**Comment by eorojas** [comment]
Yeah I don't have the bandwidth Frank.
But I'd love to chat with you about your design and see if I can help I
have a lot of experience in this domain.
